py/sentry_relay/processing.py

```diff
@@ -204,3 +205,17 @@ def parse_release(release):
     return json.loads(
         decode_str(rustcall(lib.relay_parse_release, encode_str(release)), free=True)
     )
+
+
+def validate_dynamic_rule_condition(condition):
```

Comment:
Can we validate the entire `SamplingConfig` object instead of just the condition? Otherwise, there could be errors outside of the rules.

Reply:
We certainly can. 
I thought about this for a bit. 
In the end I decided that the best thing to do is validate the most "volatile" part in Rust, and the rest in Python.
The reason for this is that using the DjangoRest validators gives a very precise location and structure to the validation error. I could replicate this structure in Rust but it would seem strange and disconnected (and some additional work).

So to get the best of both worlds: precise location of error and the confidence that the validator is kept in sync, I opted for this "mixed" approach.

Reply:
Makes sense, using Django validators for precise location is good. You could still have librelay validate the entire configuration structure to ensure we never pass invalid data on (especially when we make updates). The Python code in Sentry is too likely to go out of sync. Having both side-by-side should also work.

Reply:
Ok added a global validator (to validate the whole structure with a Yes/No , a condition validator to pinpoint problem conditions and will be using both in DjangoRest validators to get nice error messages